### ISSUE TITLE
Job clusters not appearing in cluststatefact unless Overwatch tables + data deleted

### ISSUE THREAD

**Issue by j0nnyr0berts** [databrickslabs/overwatch#148]
I'm trying to use Overwatch to monitor costs at a hourly rate however I can only get the full amount of job clusters to appear by dropping all Overwatch databases and deleting all Overwatch files.

Attached are two csvs with the Overwatch output for yesterday (28/06/21), one before and one after dropping all Overwatch tables. The row count goes from 90 -> 644. The difference appears to be the inclusion of many more 'jobs' cluster entries. I don't think think this is related to logs syncing as I experienced the same thing over the last 4 days - there was a dramatic increase in entries only after dropping all the tables and re-running. N.b. the count of jobs only matches the count on the  Accounts / Usage page after rerunning. 

I suppose my current workaround to get a view of today's spend would be to drop the Overwatch databases and delete all the data before each run, reducing the lookback to 1 day to reduce cost.

[pre_drop.csv](https://github.com/databrickslabs/overwatch/files/6732649/pre_drop.csv)
[post_drop.csv](https://github.com/databrickslabs/overwatch/files/6732648/post_drop.csv)

**Comment by GeekSheikh** [comment]
job costs have already been calculated in `jobruncostpotentialfact`. You'll need to smooth jobs running more than one day `running_dates` and if you want it by hour, you'll need to apply standard hourly smoothing techniques to the data.

I have no idea why you are dropping and reloading tables. You should just be running the Overwatch job 1X per day and the new data will flow in assuming the configuration is accurate. I'm a little confused here.

**Comment by j0nnyr0berts** [comment]
I am dropping the tables because it appears to be the only way I can get Overwatch to display the correct number of jobs  - running without dropping the tables sees jobs undercounted by as much as 10x. Perhaps I have got the config incorrect, however no error messages show so I may need you to point me in the right direction.

I'm not sure what you mean with regard to smoothing, I'm not really interested in general patterns but rather an exact (as possible) calculation of spend for the current day so that we may trigger alerts when over a threshold - perhaps that is out of scope for Overwatch?

Thanks again for your quick response.

**Comment by j0nnyr0berts** [comment]
A more succinct wording of my initial question could be, why do I only get the output in `post_drop.csv ` after dropping the Overwatch tables. Repeated runs of Overwatch without dropping them give me `pre_drop.csv` which omits much of the desired data.

**Comment by j0nnyr0berts** [comment]
Any update on the above? Before dropping the tables and rerunning, clusterstatefact had 342 entries for the last 7 days. After, 3527. Is there anything in the config that might explain this? I can't think of anything else to try.

**Comment by GeekSheikh** [comment]
Hi @j0nnyr0berts, sorry for the delay. To look into this I will need to see the following items. If you provide us with these we will dig in and see what's causing your challenge.

- Config, if you can post a redacted `JsonUtils.compactString` ([from example runner notebook](https://databrickslabs.github.io/overwatch/assets/GettingStarted/aws_runner_docs_example_042.html))
- PipReport export (final command in above example runner notebook) AFTER several subsequent runs that are resulting in you not receiving the updated data.
- All logs (zip of all log4j and stdout) for the runs, if you can label the file names a bit that will help

**Comment by GeekSheikh** [comment]
Hi @j0nnyr0berts, were you able to get this resolved? Should we close this ticket?

Thanks

**Comment by GeekSheikh** [comment]
Please let us know if you'd like us to re-open this, closing for now.

**Comment by j0nnyr0berts** [comment]
Thanks for getting back to me and sorry for the delay - have been on holiday! I will generate the required logs today.

**Comment by j0nnyr0berts** [comment]
Requested files attached
[overwatch_logs.zip](https://github.com/databrickslabs/overwatch/files/6857071/overwatch_logs.zip)
